spec/queries/subscriber_list_query_spec.rb

```diff
@@ -1,6 +1,7 @@
 RSpec.describe SubscriberListQuery do
```

Comment:
Could you add a test for `content_id: ""`?  That's a valid thing which can be sent to the API, based on the docs, so we need to make sure that behaves the same as `content_id: nil`.

Reply:
[How's this?](https://github.com/alphagov/email-alert-api/pull/1640/commits
/3f918b552120f3bd473f7e7e6dd0aaa04cb43d67#diff-fae1d811d2f95c3d665906fe8e504e659a3026697b594b5e3aeb5b5607f2da69R16)

`it { is_included_in_links tags_or_links, content_id: nil}` returns the same so they should be the same.
however as nil is in the defaults i don't think we just stick with this new L16

Reply:
Yeah, that's good

Reply:
Aha, I see, `""` is turned into `nil` when matching a UUID field:

```
irb(main):003:0> puts SubscriberList.where(content_id: [nil, ""]).to_sql
SELECT "subscriber_lists".* FROM "subscriber_lists" WHERE ("subscriber_lists"."content_id" = NULL OR "subscriber_lists"."content_id" IS NULL)
```

Looks like the same happens for any blank value:

```
irb(main):006:0> puts SubscriberList.where(content_id: [false, "     "]).to_sql
SELECT "subscriber_lists".* FROM "subscriber_lists" WHERE "subscriber_lists"."content_id" IN (NULL)
```

Reply:
Hmm, that doesn't actually look like correct behaviour from ActiveRecord though, because `NULL = NULL` is false in SQL (SQL uses a 3-valued logic):

```ruby
irb(main):010:0> SubscriberList.create!(title: "test", slug: "slug")

irb(main):015:0> pp SubscriberList.where(content_id: "").to_a ; nil
  SubscriberList Load (0.4ms)  SELECT "subscriber_lists".* FROM "subscriber_lists" WHERE "subscriber_lists"."content_id" = $1  [["content_id", nil]]
[]

irb(main):018:0> pp SubscriberList.where(content_id: nil).to_a ; nil
  SubscriberList Load (0.4ms)  SELECT "subscriber_lists".* FROM "subscriber_lists" WHERE "subscriber_lists"."content_id" IS NULL
[#<SubscriberList:0x00005617a3f12398
  id: 2,
  title: "test",
  created_at: Fri, 25 Jun 2021 13:18:18.062446000 BST +01:00,
  updated_at: Fri, 25 Jun 2021 13:18:18.062446000 BST +01:00,
  document_type: "",
  tags: {},
  links: {},
  email_document_supertype: "",
  government_document_supertype: "",
  signon_user_uid: nil,
  slug: "slug",
  url: nil,
  tags_digest: nil,
  links_digest: nil,
  content_id: nil>]

irb(main):019:0> pp SubscriberList.where(content_id: ["", "     "]).to_a ; nil
  SubscriberList Load (0.4ms)  SELECT "subscriber_lists".* FROM "subscriber_lists" WHERE "subscriber_lists"."content_id" IN (NULL)
[]
```

Reply:
I think there's another change we need to make for `FindExactQuery` (see other comment)

Reply:
I'll open an issue on Rails to ask about why it's trying to compare `NULL`s for equality

Reply:
nb: https://github.com/rails/rails/issues/42600